meson.build

```diff
@@ -0,0 +1,330 @@
+project(
+  'exiv2',
+  'cpp',
+  version: '1.0.0',
+  meson_version: '>=0.49.0',
+  default_options: ['warning_level=1', 'cpp_std=c++17'],
+)
+
+cargs = []
+cpp = meson.get_compiler('cpp')
+if host_machine.system() == 'windows' and get_option('default_library') != 'static'
+  cargs += '-DEXIV2API=__declspec(dllexport)'
+elif cpp.has_function_attribute('visibility')
+  cargs += '-DEXIV2API=__attribute__((visibility("default")))'
+else
+  cargs += '-DEXIV2API='
+endif
+
+if host_machine.system() == 'windows'
+  if cpp.get_argument_syntax() == 'gcc'
+    add_project_arguments('-D__USE_MINGW_ANSI_STDIO', '-D__MINGW_USE_VC2005_COMPAT', language: 'cpp')
```

Comment:
Any particular reason for controlling these directly? IMHO they should be left as determined by the MSYS2/MinGW-w64 toolchain.

Reply:
Another platform we should probably stop supporting on main ;) Fine though...

Reply:
MSVC32bit since 2005 does 64-bit time_t btw.